tests/visualization_tests/test_param_importances.py

```diff
@@ -9,76 +13,121 @@
-    )  # "param_a", "param_c" are conditional.
-    assert math.isclose(1.0, sum(i for i in figure.data[0].x), abs_tol=1e-5)
-
-    # Test with a customized target name.
-    figure = plot_param_importances(study, target_name="Target Name")
-    assert figure.layout.xaxis.title.text == "Importance for Target Name"
+    study.optimize(lambda t: t.suggest_float("x", 0, 1), n_trials=1)
 
-    # Test with wrong parameters.
     with pytest.raises(ValueError):
-        plot_param_importances(study, params=["optuna"])
+        _get_importances_info(study, None, None, None, "Objective Value")
 
-    # Ignore failed trials.
-    study = create_study()
-    study.optimize(fail_objective, n_trials=1, catch=(ValueError,))
-    figure = plot_param_importances(study)
-    assert len(figure.data) == 0
+
+@parametrize_plot_param_importances
+def test_plot_param_importances_customized_target_name(
+    plot_param_importances: Callable[..., Any]
+) -> None:
+
+    params = ["param_a", "param_b"]
+    study = prepare_study_with_trials()
+    figure = plot_param_importances(study, params=params, target_name="Target Name")
+    if isinstance(figure, go.Figure):
+        assert figure.layout.xaxis.title.text == "Importance for Target Name"
+    elif isinstance(figure, Axes):
+        assert figure.figure.axes[0].get_xlabel() == "Importance for Target Name"
+
+
+@parametrize_plot_param_importances
+@pytest.mark.parametrize(
+    "specific_create_study",
+    [
+        create_study,
+        _create_study_with_failed_trial,
+        prepare_study_with_trials,
+    ],
+)
+@pytest.mark.parametrize(
+    "params",
+    [
+        [],
+        ["param_a"],
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+        ["param_a", "param_b", "param_c", "param_d"],
+        None,
+    ],
+)
+def test_plot_param_importances(
+    plot_param_importances: Callable[..., Any],
+    specific_create_study: Callable[[], Study],
+    params: Optional[List[str]],
+) -> None:
+    study = specific_create_study()
+    figure = plot_param_importances(study, params=params)
+    if isinstance(figure, go.Figure):
+        figure.write_image(BytesIO())
+    else:
+        plt.savefig(BytesIO())
+
+
+@pytest.mark.parametrize(
+    "specific_create_study",
+    [create_study, _create_study_with_failed_trial],
+)
+@pytest.mark.parametrize(
+    "params",
+    [
+        [],
+        ["param_a"],
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+        ["param_a", "param_b", "param_c", "param_d"],
+        None,
+    ],
+)
+def test_get_param_importances_info_empty(
+    specific_create_study: Callable[[], Study], params: Optional[List[str]]
+) -> None:
+
+    study = specific_create_study()
+    info = _get_importances_info(
+        study, None, params=params, target=None, target_name="Objective Value"
+    )
+    assert len(info.importance_values) == 0
+    assert len(info.param_names) == 0
+    assert len(info.importance_labels) == 0
```

Comment:
How about comparing it directly with an `_ImportancesInfo`?
```suggestion
    assert info == _ImportancesInfo(importance_values=[], param_names=[], importance_labels=[], target_name="Objective Value")
```